### PR TITLE
Introduction of basic CUDA builds and tests on lassen

### PR DESCRIPTION
**Pull request by adrienbernede**
This branch adds testing of a cuda build on lassen. Dependencies are installed with Spack through uberenv.
A .mk config file is generated to point to the dependencies and apply the correct options to a subsequent build on MFEM outside of Spack.

This PR relates to https://github.com/mfem/mfem-uberenv/pull/1

- [x] The change in mfem-uberenv should be merged first.
- [x] Then the commit hash of the merge should be updated in tests/gitlab/get_mfem_uberenv

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1367](https://github.com/mfem/mfem/pull/1367) | @adrienbernede | @tzanio | @tzanio + @v-dobrev | 04/29/21 | 05/04/21 | 05/05/21 | |

### CODE REVIEW

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
LGTM

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by adrienbernede** [comment]
LGTM

**Review by tzanio — APPROVED** [review]
Thanks @adrienbernede !

File: .gitlab/lassen.yml [hunk 1]
```diff
@@ -15,43 +15,19 @@
...
-  extends: [.with_gcc_8_3_1, .on_lassen]
-  needs: [setup]
-  stage: lassen_build
-  script:
-    - mkdir -p ${BUILD_PATH}
-    - cp -r ${CI_PROJECT_DIR} ${BUILD_PATH}/${CI_PROJECT_NAME}_lassen_ser
-    - cd ${BUILD_PATH}/${CI_PROJECT_NAME}_lassen_ser
-    - lalloc 1 -W 5 -q pdebug make -j cuda CUDA_ARCH=sm_70
+  rules:
+    - if: '$CI_COMMIT_BRANCH =~ /_lnone/ || $ON_LASSEN == "OFF"' #run except if ...
```
Comment (v-dobrev) [inline .gitlab/lassen.yml:19]: When is `$CI_COMMIT_BRANCH` going to be `/_lnone/`?

Comment (adrienbernede) [reply .gitlab/lassen.yml:19]: This is just a way for me, or developers, not to trigger the tests on lassen if for some reason they don't want to:
Just append "_lnone" to your branch name.

Comment (v-dobrev) [reply .gitlab/lassen.yml:19]: In that case, shouldn't the pattern be `.*_lnone` instead of `/_lnone/`?

**Review by v-dobrev — APPROVED** [review]

File: .gitlab/lassen.yml (hunk 1 above)
Comment (adrienbernede) [reply .gitlab/lassen.yml:19]: It works well that way. Plus, if I don't want any CI, I can say "_qnone_lnone".

**Comment by tzanio** [comment]
Merged in `next` for testing...

File: .gitlab/lassen.yml (hunk 1 above)
Comment (v-dobrev) [reply .gitlab/lassen.yml:19]: Isn't this parsed in bash with `[[ condition ]]`? I don't think `/_lnone/` defines a pattern (that matches strings containing `_lnone`) in bash.

Comment (adrienbernede) [reply .gitlab/lassen.yml:19]: For reference:
https://docs.gitlab.com/ee/ci/variables/README.html#syntax-of-cicd-variable-expressions
following RE2 regular expression syntax (https://github.com/google/re2/wiki/Syntax).

Comment (v-dobrev) [reply .gitlab/lassen.yml:19]: I see. I didn't expect that.

**Pull request merged**